be/src/olap/rowset/segment_v2/binary_dict_page.cpp

```diff
@@ -45,17 +46,27 @@ BinaryDictPageBuilder::BinaryDictPageBuilder(const PageBuilderOptions& options)
           _finished(false),
           _data_page_builder(nullptr),
           _dict_builder(nullptr),
-          _encoding_type(DICT_ENCODING) {
+          _encoding_type(DICT_ENCODING) {}
+
+Status BinaryDictPageBuilder::init() {
+    RETURN_IF_ERROR_OR_CATCH_EXCEPTION(_init());
```

Comment:
add a flag to indicate init is called
and add dcheck to all method to check other method is called after init.